ant-networking/src/cmd.rs

```diff
@@ -551,6 +549,26 @@ impl SwarmDriver {
                         trace!("Replicate cmd to self received, ignoring");
                     }
                 } else {
+                    // dial the peer and send the request
+                    if addrs.0.is_empty() {
+                        info!("No addresses for peer {peer:?} to send request. This could cause dial failure if swarm could not find the peer's addrs.");
+                    } else {
+                        let opts = DialOpts::peer_id(peer)
```

Comment:
shall we carry out the additional dial `AFTER` got an error ?
i.e. make dial a separate driver cmd, and to be executed on network level, instead of within the driver instance

Reply:
> shall we carry out the additional dial AFTER got an error ?

yeah sounds good. 

> i.e. make dial a separate driver cmd, and to be executed on network level, instead of within the driver instance

Dial is not blocking, it just adds the item to the queue for the swarm, so we should be fine? moving it to the networking layer could create a lot of indirections which seems unnecessary for this instance?